cpp/src/lance/format/schema.h

```diff
@@ -138,6 +138,10 @@ class Field final {
 
   std::string logical_type() const { return logical_type_; };
 
+  std::string extension_name() const { return extension_name_; }
```

Comment:
prob can make the method to `const std::string& extension_name()` here.

Reply:
was trying to stay consistent with logical_type and name. Any particular reason this should be different?

Reply:
Oh yea, lets change logical type to `const string& `, avoid a copy each time calling the string.